tests/integration/kubeflags/kubeflags_test.go

```diff
@@ -108,10 +108,60 @@ var _ = Describe("create a new cluster with kube-* flags", Ordered, func() {
 					return nil
 				}
 				return errors.New("error finding kube-proxy")
-			}, "120s", "15s").Should(Succeed())
+			}, "300s", "15s").Should(Succeed())
 		})
+		It("kills previous server, clean logs and start up with no problems and disabled kube-proxy and cloud and cloud controller", func() {
+			var err error
+			Expect(testutil.K3sKillServer(server)).To(Succeed())
+			localServerArgs := []string{
+				"--cluster-init",
+				"--disable-cloud-controller=true",
+				"--disable-kube-proxy=true",
+			}
+			server, err = testutil.K3sStartServer(localServerArgs...)
+			Expect(err).ToNot(HaveOccurred())
 
+			// Pods should not be healthy without kube-proxy
+			Consistently(func() error {
+				return testutil.K3sDefaultDeployments()
+			}, "100s", "5s").Should(HaveOccurred())
+		})
+		It("should not find kube-proxy starting", func() {
+			Consistently(func() error {
+				match, err := testutil.SearchK3sLog(server, "Running kube-proxy")
+				if err != nil {
+					return err
+				}
+				if !match {
+					return nil
+				}
+				return errors.New("found kube-proxy starting")
+			}, "100s", "5s").Should(Succeed())
+		})
+		/* The flag --disable-cloud-controller doesn't stop ccm from running,
+		it appends -cloud-node and -cloud-node-lifecycle to the end of the --controllers flag
+		https://github.com/k3s-io/k3s/blob/master/docs/adrs/servicelb-ccm.md
```

Comment:
Another note on this - if you use --disable-cloud-controller and --disable=servicelb together, the CCM **will** be entirely disabled and prevented from running at all. You might want to cover this case in or alongside this test.